### PR TITLE
Update cleanEnumCache to be compatible with Eclipse's OpenJ9 JVM.

### PR DESCRIPTION
**Pull request by clienthax**
https://github.com/MinecraftForge/MinecraftForge/pull/5712

### CODE REVIEW

**Comment by liach** [comment]
hmm, we need an inference system so that this can work with other jvm implementations like corretto.

**Comment by clienthax** [comment]
> 
> 
> hmm, we need an inference system so that this can work with other jvm implementations like corretto.

Corretto seems to follow the same vars as oracle for now
https://github.com/corretto/corretto-8/blob/79d2ce063acaa09607109750115c3e5a33f1e4e3/src/jdk/src/share/classes/java/lang/Class.java#L3380

File: src/main/java/net/minecraftforge/fml/unsafe/UnsafeHacks.java [hunk 1]
```diff
@@ -97,5 +97,6 @@ public static void cleanEnumCache(Class<? extends Enum<?>> enumClass) throws Exc
     {
         findField(Class.class, "enumConstantDirectory").ifPresent(f -> setField(f, enumClass, null));
         findField(Class.class, "enumConstants").ifPresent(f -> setField(f, enumClass, null));
+        findField(Class.class, "enumVars").ifPresent(f -> setField(f, enumClass, null));//Open J9
```
Comment (hYdos) [inline src/main/java/net/minecraftforge/fml/unsafe/UnsafeHacks.java:100]: the other lines of code here dont have comments. this one doesnt really need one either.

**Comment by LexManos** [comment]
If this can be made generic to not have to use special internals names that'd be the best route.
However, as it sits, Forge/Unsafe are not ment to run on OpenJ9 due to it having TONS of other internal inconsistency that make what we do with classloading and the like not work well.

File: src/main/java/net/minecraftforge/fml/unsafe/UnsafeHacks.java (hunk 1 above)
Comment (clienthax) [reply src/main/java/net/minecraftforge/fml/unsafe/UnsafeHacks.java:100]: The other lines of code assume a standard JDK, this one does not.

**Pull request closed without merging**